PianoFromAbove/GameState.cpp

```diff
@@ -2063,41 +1948,144 @@ void MainScreen::RenderStatusLine(int line, const char* left, const char* format
+        RenderStatusLine(cur_line++, "Score:", "%07.0f", floor(static_cast<float>(m_iNoteCount) / static_cast<float>(mInfo.iNoteCount) * 1000000));
+             if (mInfo.iNoteCount < 80000) {RenderStatusLine(cur_line++,"Level:", "EZ Lv.1");}
+        else if (mInfo.iNoteCount < 100000) {RenderStatusLine(cur_line++,"Level:", "EZ Lv.2");}
+        else if (mInfo.iNoteCount < 200000) {RenderStatusLine(cur_line++,"Level:", "EZ Lv.3");}
+        else if (mInfo.iNoteCount < 400000) {RenderStatusLine(cur_line++,"Level:", "EZ Lv.4");}
+        else if (mInfo.iNoteCount < 600000) {RenderStatusLine(cur_line++,"Level:", "EZ Lv.5");}
+        else if (mInfo.iNoteCount < 800000) {RenderStatusLine(cur_line++,"Level:", "HD Lv.6");}
+        else if (mInfo.iNoteCount < 1000000) {RenderStatusLine(cur_line++,"Level:", "HD Lv.7");}
+        else if (mInfo.iNoteCount < 2000000) {RenderStatusLine(cur_line++,"Level:", "HD Lv.8");}
+        else if (mInfo.iNoteCount < 4000000) {RenderStatusLine(cur_line++,"Level:", "HD Lv.9");}
+        else if (mInfo.iNoteCount < 6000000) {RenderStatusLine(cur_line++,"Level:", "HD Lv.10");}
+        else if (mInfo.iNoteCount < 8000000) {RenderStatusLine(cur_line++,"Level:", "IN Lv.11");}
+        else if (mInfo.iNoteCount < 10000000) {RenderStatusLine(cur_line++,"Level:", "IN Lv.12");}
+        else if (mInfo.iNoteCount < 20000000) {RenderStatusLine(cur_line++,"Level:", "IN Lv.13");}
+        else if (mInfo.iNoteCount < 40000000) {RenderStatusLine(cur_line++,"Level:", "IN Lv.14");}
+        else if (mInfo.iNoteCount < 60000000) {RenderStatusLine(cur_line++,"Level:", "IN Lv.15");}
+        else if (mInfo.iNoteCount < 80000000) {RenderStatusLine(cur_line++,"Level:", "AT Lv.16");}
+        else if (mInfo.iNoteCount < 100000000) {RenderStatusLine(cur_line++,"Level:", "AT Lv.17");}
+        else if (mInfo.iNoteCount < 200000000) {RenderStatusLine(cur_line++,"Level:", "AT Lv.18");}
+        else if (mInfo.iNoteCount < 400000000) {RenderStatusLine(cur_line++,"Level:", "SP Lv.?");}
+        RenderStatusLine(cur_line++, "AutoPlay:", "ON ");
+        if (static_cast<float>(floor(static_cast<float>(m_iNoteCount) / static_cast<float>(mInfo.iNoteCount) * 1000000)) == static_cast<float>(1000000)) {
+            RenderStatusLine(cur_line++,"FULL COMBO!!! ", "");
+        }
+        else if(static_cast<float>(floor(static_cast<float>(m_iNoteCount) / static_cast<float>(mInfo.iNoteCount) * 1000000)) < static_cast<float>(1000000)) {
+            if (!cPlayback.GetPaused()) {
+                RenderStatusLine(cur_line++,"Deomonstration in progress... ", "");
+            }
+            else {
+                RenderStatusLine(cur_line++, "Game paused... ", "");
+            }
+        }
+        else if (static_cast<float>(floor(static_cast<float>(m_iNoteCount) / static_cast<float>(mInfo.iNoteCount) * 1000000)) > static_cast<float>(1000000)) {
+            RenderStatusLine(cur_line++,"ERROR: Score overflow! ", "");
+        }
```

Comment:
Feel free to keep something like this in your fork, but I don't think this is something I'd merge.